### ISSUE TITLE
Nuxi Init fails when using NPM (@parcel/watcher)

### ISSUE THREAD

**Issue by scalemaildev** [nuxt/nuxt#25904]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v21.0.0
- Nuxt Version:     -
- CLI Version:      3.10.1
- Nitro Version:    -
- Package Manager:  unknown
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

Running "npx nuxi@latest init repo' in a terminal

### Describe the bug

Selecting NPM as package manager during nuxi init (on latest version of node/npm/nuxi) throws error during dependency setup.

I was previously able to init Nuxt3 repos just fine. Only change I can think of was that I recently upgraded NPM. Using Yarn works just fine however, as does using npm install after successful Yarn setup.

### Logs

```shell-script
`4440 verbose Linux 5.15.0-94-generic
4441 verbose node v21.0.0
4442 verbose npm  v10.4.0
4443 error code 1
4444 error path /home/clownway/Projects/delvinator/node_modules/@parcel/watcher
4445 error command failed
4446 error command sh -c node-gyp rebuild
4447 error gyp info it worked if it ends with ok
4447 error gyp info using node-gyp@10.0.1
4447 error gyp info using node@21.0.0 | linux | x64
4447 error gyp info find Python using Python version 3.10.12 found at "/usr/bin/python3"
4447 error gyp info spawn /usr/bin/python3
4447 error gyp info spawn args [
4447 error gyp info spawn args '/home/clownway/Projects/delvinator/node_modules/node-gyp/gyp/gyp_main.py',
4447 error gyp info spawn args 'binding.gyp',
4447 error gyp info spawn args '-f',
4447 error gyp info spawn args 'make',
4447 error gyp info spawn args '-I',
4447 error gyp info spawn args '/home/clownway/Projects/delvinator/node_modules/@parcel/watcher/build/config.gypi',
4447 error gyp info spawn args '-I',
4447 error gyp info spawn args '/home/clownway/Projects/delvinator/node_modules/node-gyp/addon.gypi',
4447 error gyp info spawn args '-I',
4447 error gyp info spawn args '/home/clownway/.cache/node-gyp/21.0.0/include/node/common.gypi',
4447 error gyp info spawn args '-Dlibrary=shared_library',
4447 error gyp info spawn args '-Dvisibility=default',
4447 error gyp info spawn args '-Dnode_root_dir=/home/clownway/.cache/node-gyp/21.0.0',
4447 error gyp info spawn args '-Dnode_gyp_dir=/home/clownway/Projects/delvinator/node_modules/node-gyp',
4447 error gyp info spawn args '-Dnode_lib_file=/home/clownway/.cache/node-gyp/21.0.0/<(target_arch)/node.lib',
4447 error gyp info spawn args '-Dmodule_root_dir=/home/clownway/Projects/delvinator/node_modules/@parcel/watcher',
4447 error gyp info spawn args '-Dnode_engine=v8',
4447 error gyp info spawn args '--depth=.',
4447 error gyp info spawn args '--no-parallel',
4447 error gyp info spawn args '--generator-output',
4447 error gyp info spawn args 'build',
4447 error gyp info spawn args '-Goutput_dir=.'
4447 error gyp info spawn args ]
4447 error gyp: binding.gyp not found (cwd: /home/clownway/Projects/delvinator/node_modules/@parcel/watcher) while trying to load binding.gyp
4447 error gyp ERR! configure error 
4447 error gyp ERR! stack Error: `gyp` failed with exit code: 1
4447 error gyp ERR! stack at ChildProcess.<anonymous> (/home/clownway/Projects/delvinator/node_modules/node-gyp/lib/configure.js:271:18)
4447 error gyp ERR! stack at ChildProcess.emit (node:events:515:28)
4447 error gyp ERR! stack at ChildProcess._handle.onexit (node:internal/child_process:294:12)
4447 error gyp ERR! System Linux 5.15.0-94-generic
4447 error gyp ERR! command "/home/clownway/.nvm/versions/node/v21.0.0/bin/node" "/home/clownway/Projects/delvinator/node_modules/.bin/node-gyp" "rebuild"
4447 error gyp ERR! cwd /home/clownway/Projects/delvinator/node_modules/@parcel/watcher
4447 error gyp ERR! node -v v21.0.0
4447 error gyp ERR! node-gyp -v v10.0.1
4447 error gyp ERR! not ok
4448 verbose exit 1
4449 timing npm Completed in 48606ms
4450 verbose unfinished npm timer reify 1708573190152
4451 verbose unfinished npm timer reify:build 1708573237295
4452 verbose unfinished npm timer build 1708573237297
4453 verbose unfinished npm timer build:deps 1708573237298
4454 verbose unfinished npm timer build:run:install 1708573237346
4455 verbose unfinished npm timer build:run:install:node_modules/@parcel/watcher 1708573237346
4456 verbose code 1`
```

**Comment by hiddenLadder** [comment]
same

**Comment by danielroe** [comment]
This is an upstream issue - you can track in https://github.com/parcel-bundler/watcher/issues/156.

For now, you could downgrade the library in your project following this example: https://github.com/aws-amplify/amplify-backend/pull/1045/files

**Comment by mcfarljw** [comment]
It only worked for me when using `overrides` in `package.json`:

```
"overrides": {
  "@parcel/watcher": "~2.1.0"
}
```

It seems like it's a conflict between the latest versions of @parcel/watcher and npm@10.4.0. Luckily that version of npm isn't installed by default with node but if anyone runs `npm -g update` they'll likely run into this issue and need to override the @parcel/watcher version.

**Comment by devongovett** [comment]
Please try `@parcel/watcher` v2.4.1, which should fix this issue with npm 10.4.0. See https://github.com/parcel-bundler/watcher/issues/156#issuecomment-1961552001 for details.
